BUGFIX: site list in "content" of left drawer policy checked

tdausner opened the pull request:
Using policies for NodeTreePrivilege (in Policy.yaml) did not reflect in the list of sites in left drawer "content" site list. Now a restricted user only sees sites with NodeTreePrivilege access GRANTED.

Function buildSiteList is enriched by $context and $privilegeManager. On each site the node of path `/sites/{root-node-name-of-site}` is checked by `$privilegeManager->isGranted()` and put into site list on GRANTED.

**Use case / Review instructions**

- start with a fresh NEOS development installtion (see https://github.com/neos/neos-development-collection#contributing)
- kickstart multi site setup (First.Site, Second.Site, Third.Site) by command 
`$ for s in First Second Third; do ./flow kickstart:site $s.Site $s ; done`

- Import sites just created
`$ for s in First Second Third; do ./flow site:import --package-key $s.Site ; done`

- domain setup for sites in backed Administration/Sites Management
First.Site -> https://first.tds (_tds_ is my development domain)
Second.Site -> https://second.tds
Third.Site -> https://third.tds
![image](https://user-images.githubusercontent.com/19776511/205447369-7406e6b5-5bcc-48bc-8b22-5bf9fe2cd8da.png)

- Setup `Policy.yaml` (see [Policy.yaml.txt](https://github.com/neos/neos-development-collection/files/10146777/Policy.yaml.txt) ) and check resulting roles by command
```
$ ./flow security:listroles
+-----------------------------------------------+-----------------------+
| Id                                            | Label                 |
+-----------------------------------------------+-----------------------+
| Neos.Neos:LivePublisher                       | Live publisher        |
| Neos.Neos:Administrator                       | Neos Administrator    |
| Neos.Neos:RestrictedEditor                    | Restricted Editor     |
| Neos.Neos:Editor                              | Editor                |
| Neos.Neos:UserManager                         | Neos User Manager     |
| Neos.RedirectHandler.Ui:RedirectAdministrator | RedirectAdministrator |
| Neos.Setup:SetupUser                          | SetupUser             |
| First.Site:FirstEditor                        | First site Editor     |
| Second.Site:SecondEditor                      | Second site Editor    |
| Third.Site:ThirdEditor                        | Third site Editor     |
+-----------------------------------------------+-----------------------+
```
`NodeTreePrivilege` matcher in `Policy.yaml` is `'isDescendantNodeOf("/sites/first-site")'` for First.Site and so on. Retrieve the node names by command:
```
$ ./flow site:list

 Name                 Node name                 Resources package               Status
----------------------------------------------------------------------------------------
 First                first-site                First.Site                      online
 Second               second-site               Second.Site                     online
 Third                third-site                Third.Site                      online

```
- Add test user:
`$ ./flow user:create --roles Second.Site:SecondEditor TestUser testuser-password Test User`

- Log into NEOS as administrator
![image](https://user-images.githubusercontent.com/19776511/205448906-427e46af-ec1f-481f-9e06-c9b212426dc8.png)

- Log into NEOS as TestUser
![image](https://user-images.githubusercontent.com/19776511/205448973-b80e5c70-10e4-47ca-8e67-1a1d9de220ac.png)

**Checklist**

- [x] Code follows the PSR-2 coding style
- [x] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

tdausner committed:
BUGFIX: site list in "content" of left drawer policy checked

Using policies for NodeTreePrivilege (in Policy.yaml) did not reflect in the list of sites in left drawer "content" site list. Now a restricted user only sees sites with NodeTreePrivilege access GRANTED.

markusguenther commented:
@tdausner Thanks for this contribution. Does this only happen on Neos 8.2?
Otherwise, we can target another version where this also occurs. Need to check that in a calm moment.

tdausner commented:
@markusguenther had a look at 7.0 and 5.0 releases. Both versions don't pay respect to policies so IMHO this did occur to any version in the past.

tdausner committed:
BUGFIX: site list in "content" of left drawer policy checked

Utilization of $this->privilegeManager (no $privilegeManager initialization in function buildSiteList)

tdausner committed:
BUGFIX: site list in "content" of left drawer policy checked

reduction unused declarations

tdausner committed:
Merge branch 'neos:8.2' into bugfix/site.list-in-left-drawer

tdausner committed:
Revert "Merge branch 'neos:8.2' into bugfix/site.list-in-left-drawer"

This reverts commit 2793e9969421f7f118bb9124bc5cb831c88e7e99, reversing
changes made to 3dda76b6e75800d0841f7a8b71004280bb156538.
